File: in_Java/1260 Shift 2D Grid.java
class Solution {
    public List<List<Integer>> shiftGrid(int[][] grid, int k) {
        int row = grid.length;
        int col = grid[0].length;
        int[] t = new int[row*col];
        int id = 0;
        for(int i = 0; i < row; i++){
            for(int j = 0; j < col; j++){
                t[id++] = grid[i][j];
            }
        }
        k = k % t.length;
        int[] tt = new int[t.length];        
        for(int i = 0; i < k; i++){
            tt[i] = t[t.length-k+i];
        }
        for(int i = k; i < t.length; i++){
            tt[i] = t[i-k];
        }
        id = 0;
        List<List<Integer>> res = new ArrayList<>();
        for(int i = 0; i < row; i++){
            List<Integer> cur = new ArrayList<>();
            for(int j = 0; j < col; j++){
                cur.add(tt[id++]);
            }
            res.add(cur);
        }
        return res;
    }
}

File: in_Python/1260 Shift 2D Grid.py
class Solution(object):
    def shiftGrid(self, grid, k):
        """
        :type grid: List[List[int]]
        :type k: int
        :rtype: List[List[int]]
        """
        row = len(grid)
        col = len(grid[0])
        while k != 0:
            k -= 1
            lastcol = []
            for i in range(row):
                lastcol.append(grid[i][col-1])
            for i in range(row):
                for j in range(col-1, 0, -1):
                    grid[i][j] = grid[i][j-1]
            grid[0][0] = lastcol[-1]
            idx = 0
            for i in range(1, row):
                grid[i][0] = lastcol[idx]
                idx += 1
        return grid